src/imageproxy.rs

```diff
@@ -701,11 +701,12 @@ mod tests {
         };
         config.skopeo_cmd = Some(Command::new("no-skopeo"));
 
-        let res = ImageProxy::new_with_config(config).await;
-        assert!(matches!(res, Err(Error::SkopeoSpawnError(_))));
-        assert_eq!(
-            res.err().unwrap().to_string(),
-            "skopeo spawn error: No such file or directory (os error 2)"
```

Comment:
Isn't there value in checking for the error string in the test? Part of the reason why this change was done is to ensure that the actual culprit (skopeo missing) is to see so it seems useful to have that as part of the test?

Reply:
The error strings are derived from the error types - so they should be equivalent right?

As I said in the commit message we'd expect callers to match against the types, not strings.

It wouldn't hurt at all to do both here of course (though one thing to note is that "os error 2" could be different across OSes, and there's some attempt in the podman stack to support FreeBSD)

Reply:
Yeah, I agree, I guess I just get this warm-fuzzy-feeling seeing the error spelled out in the test and that it actually looks reasonable (also in the test-as-doc sense). But maybe its just because I don't have a good sense for rust yet (it is kinda obvious what it will look like). Anyway, my my .02¢.

Reply:
(and of course the facts the exact string is OS specific is a problem and probably enough to justify not testing for it)

Reply:
Sure, we can match against the strings too. Done.